3letterfault.py

```diff
@@ -21,6 +21,8 @@ class Row:
     rows[tlc] = Row(tlc=tlc, longDesc=faultRowDict["Long Description"],
                     shortDesc=faultRowDict["Short Description"])
 
+sorted_FaultRows = {key: value for key, value in sorted(rows.items())}
```

Comment:
Hmm I think if you want this robustly sorted, it needs to be an ordered dict. I think you would probably wanna do `sorted(rows.keys())`.

Maybe something like `sortedFaultRows = OrderedDict([(tlc, rows[tlc]) for tlc in sorted(rows.keys())])`

Reply:
Implemented! I don't actually understand what/why this works (which is due to general python knowledge lacking), but it worked! At some point, I would love to walk through what this is doing

Reply:
So the first point is that a standard dictionary isn't guaranteed to be sorted, even if it looks like it will be. An ordered dict explicitly remembers/maintains the insertion order (so it's guaranteed to be sorted). The second thing is that I'm just not sure how sorted handles tuples (items() returns a list of tuples) so I figured we could just cut out the middle man and directly sort the keys (the TLCs)

Reply:
I think what's confusing me is more just the shorthand. I don't know how to interpret a line of code with a variable in front of the for loop because I've never seen that before

Reply:
Ah ok!

I suppose the equivalent of the line I suggested would be

<img width="295" alt="image" src="https://user-images.githubusercontent.com/13070175/163636213-443a2ccd-d8df-4f82-b9fa-08ca6ddea8b8.png">

Reply:
Ah, thank you!!!! That helps alot!!